devlib/bin/scripts/shutils.in

```diff
@@ -242,26 +242,32 @@ hotplug_online_all() {
 # Misc
 ################################################################################
 
-read_tree_values() {
+read_tree_tgz_b64() {
     BASEPATH=$1
     MAXDEPTH=$2
+    TMPBASE=$3
 
     if [ ! -e $BASEPATH ]; then
         echo "ERROR: $BASEPATH does not exist"
         exit 1
     fi
 
-    PATHS=$($BUSYBOX find $BASEPATH -follow -maxdepth $MAXDEPTH)
-    i=0
-    for path in $PATHS; do
-        i=$(expr $i + 1)
-        if [ $i -gt 1 ]; then
-            break;
-        fi
+    cd $TMPBASE
+    TMPDIR=$($BUSYBOX realpath $($BUSYBOX mktemp -d XXXXXX))
```

Comment:
`TMPDIR` has a special meaning for mktemp, so we may want to use a variation of that name instead to avoid issues on repeated invocations. Quote from `man mktemp`:
```
       -p DIR, --tmpdir[=DIR]
              interpret TEMPLATE relative to DIR; if DIR is not specified, use
              $TMPDIR  if set, else /tmp.  With this option, TEMPLATE must not
              be an absolute  name;  unlike  with  -t,  TEMPLATE  may  contain
              slashes, but mktemp creates only the final component
```

If TMPDIR ends up being an env var, setting it to a non-existent folder (since we delete it) will give:
```
$ TMPDIR=x mktemp 
mktemp: failed to create file via template ‘x/tmp.XXXXXXXXXX’: No such file or directory
```

Reply:
Ah, good catch. I'll make to use another name so that we don't face weird bugs later on when/if that gets changed.